pkg/services/pluginsintegration/angularpatternsstore/store.go

```diff
@@ -0,0 +1,85 @@
+package angularpatternsstore
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/grafana/grafana/pkg/infra/kvstore"
+)
+
+type Service interface {
+	Get(ctx context.Context) (string, error)
+	Set(ctx context.Context, patterns any) error
+	GetLastUpdated(ctx context.Context) (time.Time, error)
+}
+
+var ErrNoCachedValue = errors.New("no cached value")
+
+const (
+	kvNamespace = "plugin.angularpatterns"
+
+	keyPatterns    = "angular_patterns"
+	keyLastUpdated = "last_updated"
+)
+
+// KVStoreService allows to cache GCOM angular patterns into the database, as a cache.
+type KVStoreService struct {
+	kv *kvstore.NamespacedKVStore
+}
+
+func ProvideService(kv kvstore.KVStore) Service {
+	return &KVStoreService{
+		kv: kvstore.WithNamespace(kv, 0, kvNamespace),
+	}
+}
+
+// Get returns the raw cached angular detection patterns. The returned value is a JSON-encoded string.
+// If no value is present, ErrNoCachedValue is returned.
+func (s *KVStoreService) Get(ctx context.Context) (string, error) {
+	data, ok, err := s.kv.Get(ctx, keyPatterns)
+	if err != nil {
+		return "", fmt.Errorf("kv get: %w", err)
+	}
+	if !ok {
+		return "", ErrNoCachedValue
+	}
+	return data, nil
+}
+
+// Set sets the cached angular detection patterns and the latest update time to time.Now().
+// patterns must implement json.Marshaler.
+func (s *KVStoreService) Set(ctx context.Context, patterns any) error {
+	b, err := json.Marshal(patterns)
+	if err != nil {
+		return fmt.Errorf("json marshal: %w", err)
+	}
+	if err := s.kv.Set(ctx, keyPatterns, string(b)); err != nil {
+		return fmt.Errorf("kv set: %w", err)
+	}
+	if err := s.kv.Set(ctx, keyLastUpdated, time.Now().Format(time.RFC3339)); err != nil {
+		return fmt.Errorf("kv last updated set: %w", err)
+	}
```

Comment:
nice idea to wrap `Set` with the update of `LastUpdated`. We can do that for the `Set` and `Delete` methods of the generic `store`